src/UsgsAstroLsSensorModel.cpp

```diff
@@ -2727,60 +2722,51 @@ std::string UsgsAstroLsSensorModel::constructStateFromIsd(const std::string imag
+   for (int i=0;i < numEphem; i++){
+       state["m_ephemPts"].push_back(isd.at("sensor_location").at("x").at(i));
+       state["m_ephemPts"].push_back(isd.at("sensor_location").at("y").at(i));
+       state["m_ephemPts"].push_back(isd.at("sensor_location").at("z").at(i));
+       state["m_ephemRates"].push_back(isd.at("sensor_velocity").at("x").at(i));
+       state["m_ephemRates"].push_back(isd.at("sensor_velocity").at("y").at(i));
+       state["m_ephemRates"].push_back(isd.at("sensor_velocity").at("z").at(i));
+       state["m_quaternions"].push_back(isd.at("sensor_orientation").at("w").at(i));
+       state["m_quaternions"].push_back(isd.at("sensor_orientation").at("v1").at(i));
+       state["m_quaternions"].push_back(isd.at("sensor_orientation").at("v2").at(i));
+       state["m_quaternions"].push_back(isd.at("sensor_orientation").at("v3").at(i));
    }
 
-   //state["m_ephemPts"] = isd.m_ephem_pts;
-   //state["m_ephemRates"] = isd.m_ephem_rates;
-   //state["m_quaternions"] = isd.m_quaternions;
-   state["m_parameterVals"] = json::array();
-   for (int i=0; i < 18; i++){
-       state["m_parameterVals"].push_back(isd.at("TRI_PARAMETERS").at(i));
-   }
-   //state["m_parameterVals"] = isd.m_tRI_PARAMETERS;
-   double deltaF = state["m_parameterVals"][numQuat - 1].get<double>() - state["m_focal"].get<double>();
-   if (fabs(deltaF) < 0.4 * state["m_focal"].get<double>())
-      state["m_parameterVals"][numQuat - 1] = deltaF;
+   state["m_parameterVals"] = std::vector<double>(NUM_PARAMETERS, 0.0);
+   state["m_parameterVals"][15] = state["m_focal"].get<double>();
 
    // Set the ellipsoid
-   state["m_semiMajorAxis"] = isd.at("SEMI_MAJOR_AXIS");
-   state["m_semiMinorAxis"] =
-        state["m_semiMajorAxis"].get<double>() * sqrt(1.0 - isd.at("ECCENTRICITY").get<double>() * isd.at("ECCENTRICITY").get<double>()) ;
+   state["m_semiMajorAxis"] = isd.at("radii").at("semimajor");
+   state["m_semiMinorAxis"] = isd.at("radii").at("semiminor");
 
    // Now finish setting the state data from the ISD read in
 
    // set identifiers
-   state["m_referenceDateAndTime"] = isd.at("REF_DATE_TIME");
-   state["m_platformIdentifier"]   = isd.at("PLATFORM_ID");
-   state["m_sensorIdentifier"]     = isd.at("SENSOR_ID");
-   state["m_trajectoryIdentifier"] = isd.at("TRAJ_ID");
-   state["m_collectionIdentifier"] = isd.at("COLL_ID");
+   state["m_referenceDateAndTime"] = "UNKNOWN";
+   state["m_platformIdentifier"]   = "UNKNOWN";
+   state["m_sensorIdentifier"]     = "UNKNOWN";
+   state["m_trajectoryIdentifier"] = "UNKNOWN";
+   state["m_collectionIdentifier"] = "UNKNOWN";
 
    // Ground elevations
-   state["m_refElevation"] = isd.at("REFERENCE_HEIGHT");
-   state["m_minElevation"] = isd.at("MIN_VALID_HT");
-   state["m_maxElevation"] = isd.at("MAX_VALID_HT");
+   state["m_refElevation"] = 0.0;
+   state["m_minElevation"] = isd.at("reference_height").at("minheight");
+   state["m_maxElevation"] = isd.at("reference_height").at("maxheight");

```

Comment:
I have a question on the pfeffernusse side about what to send out here. For now, I am defaulting to meaningless floats (0,100) I think.